src/shadowbox/server/manager_metrics.ts

```diff
@@ -40,4 +72,71 @@ export class PrometheusManagerMetrics implements ManagerMetrics {
     }
     return {bytesTransferredByUserId: usage};
   }
+
+  async getServerMetrics({hours}: ServerMetricsTimeframe): Promise<ServerMetrics> {
+    const dataTransferredByLocation = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_data_bytes_per_location{dir=~"c<p|p>t"}[${hours}h])) by (location, asn, asorg)`
+    );
+    const tunnelTimeByLocation = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_tunnel_time_seconds_per_location[${hours}h])) by (location, asn, asorg)`
+    );
+    const dataTransferredByAccessKey = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_data_bytes{dir=~"c<p|p>t"}[${hours}h])) by (access_key)`
+    );
+    const tunnelTimeByAccessKey = await this.prometheusClient.query(
+      `sum(increase(shadowsocks_tunnel_time_seconds[${hours}h])) by (access_key)`
+    );
+
+    const tunnelTimeByLocationMap = new Map();
+    for (const entry of tunnelTimeByLocation.result) {
+      tunnelTimeByLocationMap.set(
+        `${entry.metric['location']},${entry.metric['asn']},${entry.metric['asorg']}`,
+        parseFloat(entry.value[1])
+      );
```

Comment:
Maybe define a local `entryKey` function, so it's clear you are using the same in both loops.
```suggestion
      tunnelTimeByLocationMap.set(entryKey(entry), {seconds: parseFloat(entry.value[1])});
```